### PR TITLE
Gracefull exit fixes #4092

### LINKED ISSUE #4092: Container failures cause orphan rialto servers-RDKCOM-3921
There are two scenarios after rialto server is launched where container can fail, during intialization and during start of container. We need to ensure that rialtoserver is dismantled if that happens.

### PR DESCRIPTION
**Pull request by joseinweb**
RDKCOM-3921- Gracefull exit on failures

### CODE REVIEW

**Comment by joseinweb** [comment]
Closing this to merge with 4014 and create a new one.

File: RDKShell/RDKShell.cpp [hunk 1]
```diff
@@ -4358,19 +4358,16 @@ namespace WPEFramework {
...
+                        rialtoConnector->initialize();
                     }
                     LOGWARN("Creating app session ....");
                     if(!rialtoConnector->createAppSession(client,display, appId))
                     {
                         response["message"] = "Rialto app session initialisation failed";
                         returnResponse(false);
                     }
-                    if(rialtoConnector->waitForStateChange(appId,RialtoServerStates::INACTIVE,200))
+                    if(rialtoConnector->waitForStateChange(appId,RialtoServerStates::INACTIVE,RIALTO_TIMEOUT_MILLIS))
```
Comment (sverkoye) [inline RDKShell/RDKShell.cpp:4370]: The if test is wrong in two places. The correct if test is:

`  if(!rialtoConnector->waitForStateChange(appId,RialtoServerStates::ACTIVE, RIALTO_TIMEOUT_MILLIS))`

See my original patch: https://code.rdkcentral.com/r/c/components/generic/rdk-oe/meta-cmf-video/+/86680/1/recipes-extended/rdkservices/rdkservices/0004-RDKDEV-771-socket-wait-for-active-state.patch#13

File: RDKShell/RDKShell.cpp [hunk 2]
```diff
@@ -4546,7 +4553,7 @@ namespace WPEFramework {
                     }
 #ifdef ENABLE_RIALTO_FEATURE
                     rialtoConnector->resumeSession(client);
-                    if(rialtoConnector->waitForStateChange(client,RialtoServerStates::ACTIVE,200))
+                    if(rialtoConnector->waitForStateChange(client,RialtoServerStates::ACTIVE,RIALTO_TIMEOUT_MILLIS))
```
Comment (sverkoye) [inline RDKShell/RDKShell.cpp:4556]: The if test is wrong. The correct if test is:

`  if(!rialtoConnector->waitForStateChange(client,RialtoServerStates::ACTIVE, RIALTO_TIMEOUT_MILLIS))`

File: RDKShell/RDKShell.h [hunk 3]
```diff
@@ -28,6 +28,7 @@
 #include "tptimer.h"
 #ifdef ENABLE_RIALTO_FEATURE
 #include "RialtoConnector.h"
+#define RIALTO_TIMEOUT_MILLIS 2000
```
Comment (sverkoye) [inline RDKShell/RDKShell.h:31]: Please set the timeout to 5000 millis. We noticed problems on realtek or amlogic platforms where 2000 was too small.

**Review by sverkoye — CHANGES_REQUESTED** [review]

File: RDKShell/RDKShell.cpp (hunk 2 above)
Comment (joseinweb) [reply RDKShell/RDKShell.cpp]: Corrected

File: RDKShell/RDKShell.h (hunk 3 above)
Comment (joseinweb) [reply RDKShell/RDKShell.h]: Done

File: RDKShell/RDKShell.cpp [hunk 4]
```diff
@@ -4358,19 +4358,16 @@ namespace WPEFramework {
...
+                        rialtoConnector->initialize();
                     }
                     LOGWARN("Creating app session ....");
                     if(!rialtoConnector->createAppSession(client,display, appId))
                     {
                         response["message"] = "Rialto app session initialisation failed";
                         returnResponse(false);
                     }
-                    if(rialtoConnector->waitForStateChange(appId,RialtoServerStates::INACTIVE,200))
+                    if(!rialtoConnector->waitForStateChange(appId,RialtoServerStates::INACTIVE,RIALTO_TIMEOUT_MILLIS))
```
Comment (sverkoye) [inline RDKShell/RDKShell.cpp:4370]: Still wrong: it should wait for **RialtoServerStates::ACTIVE** state and not the **RialtoServerStates::INACTIVE** state.

**Review by sverkoye — CHANGES_REQUESTED** [review]

File: RDKShell/RDKShell.cpp (hunk 4 above)
Comment (joseinweb) [reply RDKShell/RDKShell.cpp]: Thanks for pointing it out. Fixed

**Review by binuinbaraj — APPROVED** [review]

**Review by sverkoye — APPROVED** [review]

**Review by anand-ky — APPROVED** [review]

**Pull request merged**